### ISSUE TITLE
LSP Python server never starts

### ISSUE THREAD

**Issue by pirlite2** [spyder-ide/spyder#17541]
### Issue Report Checklist

[x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
[x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
[ n/a] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
[n/a ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
[n/a ] Tried basic troubleshooting (if a bug/error)
[x ] Restarted Spyder
[x ] Reset preferences with ``spyder --reset``
[n/a ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
[x ] Tried the other applicable steps from the Troubleshooting Guide
[x ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I have installed spyder from pip using ''pip3 install spyder spyder-kernels' on a clean version of Linux Mint 20.3. I can see I have installed: spyder 5.2.2 and spyder-kernels 2.2.1

Spyder starts OK but the LSP Python server never starts. I have tried rerunning the app, restarting the server, resetting, rebooting, looking at everything I can find on web. No joy.

Edit: "The weird thing is I have a second identical machine where this worked first time." I take that back! Now I look carefully, the server is not starting. So the problem reproduces on two independent machines... 

### What steps reproduce the problem?

Starting Spyder!
 
### What is the expected output? What do you see instead?

The server starts and I never get linting, etc.!

### Paste Traceback/Error Below (if applicable)

n/a> Internal Console --->

## Versions
* Spyder version: 5.2.2 None
* Python version: 3.8.10 64-bit
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Linux 5.4.0-105-generic

### Dependencies
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  3.0.4 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0;<8.0.0        :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.4.0 (OK)
keyring >=17.0.0              :  18.0.1 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.2 (OK)
parso >=0.7.0;<0.9.0          :  0.7.1 (OK)
pexpect >=4.4.0               :  4.6.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.5.1 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0                :  2.12.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.2;<1.4.0          :  1.3.3 (OK)
pylsp_black >=1.0.0           :  1.1.0 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.2 (OK)
qtpy >=1.5.0                  :  2.0.1 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  60.10.0 (OK)
sphinx >=0.6.6                :  4.4.0 (OK)
spyder_kernels >=2.2.1;<2.3.0 :  2.2.1 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.26 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  3.5.1 (OK)
numpy >=1.7                   :  1.22.3 (OK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  1.8.0 (OK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @pirlite2, thanks for reporting. Did you use a virtualenv to install Spyder? Or did you run `sudo pip3 install spyder`?

**Comment by pirlite2** [comment]
No virtual environments. Just the default.

Originally, I installed with 'pip3 install spyder spyder-kernels'. No 'sudo'. That said, I have just tried uninstalling and re-installing with 'sudo pip3 install spyder spyder-kernels' but no change.

FWIW: 'sudo pip3 install spyder' doesn't produce any change either.

Looking at the output from 'sudo pip3 install spyder spyder-kernels', the only things that look amiss are:

...snip

ERROR: launchpadlib 1.10.13 requires testresources, which is not installed.
ERROR: jupyterlab-pygments 0.1.2 has requirement pygments<3,>=2.4.1, but you'll have pygments 2.3.1 which is incompatible.
ERROR: nbconvert 6.4.4 has requirement pygments>=2.4.1, but you'll have pygments 2.3.1 which is incompatible.
ERROR: black 22.1.0 has requirement click>=8.0.0, but you'll have click 7.0 which is incompatible.
ERROR: cookiecutter 1.7.3 has requirement requests>=2.23.0, but you'll have requests 2.22.0 which is incompatible.
Installing collected packages: atomicwrites, qtpy, qdarkstyle, cloudpickle, testpath, mistune, MarkupSafe, jinja2, jupyterlab-pygments, traitlets, python-dateutil, pyzmq, tornado, jupyter-core, nest-asyncio, jupyter-client, attrs, pyrsistent, zipp, importlib-resources, jsonschema, nbformat, nbclient, defusedxml, pandocfilters, webencodings, bleach, nbconvert, parso, ipython-genutils, matplotlib-inline, debugpy, setuptools, backcall, pickleshare, decorator, jedi, wcwidth, prompt-toolkit, ipython, ipykernel, qtconsole, PyQt5-sip, pyqt5, pyqtwebengine, inflection, tinycss2, qstylizer, diff-match-patch, three-merge, ujson, pluggy, python-lsp-jsonrpc, pycodestyle, snowballstemmer, pydocstyle, mccabe, yapf, pyflakes, flake8, dill, platformdirs, tomli, isort, typing-extensions, lazy-object-proxy, wrapt, astroid, pylint, rope, toml, autopep8, python-lsp-server, pyls-spyder, sphinxcontrib-htmlhelp, docutils, alabaster, importlib-metadata, sphinxcontrib-qthelp, sphinxcontrib-applehelp, imagesize, sphinxcontrib-serializinghtml, sphinxcontrib-jsmath, sphinxcontrib-devhelp, babel, sphinx, numpydoc, textdistance, qtawesome, wurlitzer, spyder-kernels, watchdog, jellyfish, mypy-extensions, pathspec, black, python-lsp-black, poyo, text-unidecode, python-slugify, arrow, jinja2-time, binaryornot, cookiecutter, sortedcontainers, intervaltree, rtree, spyder
  Attempting uninstall: MarkupSafe
    Found existing installation: MarkupSafe 1.1.0
    Not uninstalling markupsafe at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'MarkupSafe'. No files were found to uninstall.
  Attempting uninstall: setuptools
    Found existing installation: setuptools 45.2.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'setuptools'. No files were found to uninstall.
Successfully installed MarkupSafe-2.1.1 PyQt5-sip-12.9.1 alabaster-0.7.12 arrow-1.2.2 astroid-2.11.2 atomicwrites-1.4.0 attrs-21.4.0 autopep8-1.6.0 babel-2.9.1 backcall-0.2.0 binaryornot-0.4.4 black-22.1.0 bleach-4.1.0 cloudpickle-2.0.0 cookiecutter-1.7.3 debugpy-1.6.0 decorator-5.1.1 defusedxml-0.7.1 diff-match-patch-20200713 dill-0.3.4 docutils-0.17.1 flake8-4.0.1 imagesize-1.3.0 importlib-metadata-4.11.3 importlib-resources-5.6.0 inflection-0.5.1 intervaltree-3.1.0 ipykernel-6.9.2 ipython-7.32.0 ipython-genutils-0.2.0 isort-5.10.1 jedi-0.18.1 jellyfish-0.9.0 jinja2-3.1.1 jinja2-time-0.2.0 jsonschema-4.4.0 jupyter-client-7.1.2 jupyter-core-4.9.2 jupyterlab-pygments-0.1.2 lazy-object-proxy-1.7.1 matplotlib-inline-0.1.3 mccabe-0.6.1 mistune-0.8.4 mypy-extensions-0.4.3 nbclient-0.5.13 nbconvert-6.4.4 nbformat-5.2.0 nest-asyncio-1.5.4 numpydoc-1.2 pandocfilters-1.5.0 parso-0.8.3 pathspec-0.9.0 pickleshare-0.7.5 platformdirs-2.5.1 pluggy-1.0.0 poyo-0.5.0 prompt-toolkit-3.0.28 pycodestyle-2.8.0 pydocstyle-6.1.1 pyflakes-2.4.0 pylint-2.13.1 pyls-spyder-0.4.0 pyqt5-5.12.3 pyqtwebengine-5.12.1 pyrsistent-0.18.1 python-dateutil-2.8.2 python-lsp-black-1.1.0 python-lsp-jsonrpc-1.0.0 python-lsp-server-1.3.3 python-slugify-6.1.1 pyzmq-22.3.0 qdarkstyle-3.0.2 qstylizer-0.2.1 qtawesome-1.1.1 qtconsole-5.2.2 qtpy-2.0.1 rope-0.23.0 rtree-0.9.7 setuptools-61.1.1 snowballstemmer-2.2.0 sortedcontainers-2.4.0 sphinx-4.4.0 sphinxcontrib-applehelp-1.0.2 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 spyder-5.2.2 spyder-kernels-2.2.1 testpath-0.6.0 text-unidecode-1.3 textdistance-4.2.2 three-merge-0.1.1 tinycss2-1.1.1 toml-0.10.2 tomli-2.0.1 tornado-6.1 traitlets-5.1.1 typing-extensions-4.1.1 ujson-5.1.0 watchdog-2.1.7 wcwidth-0.2.5 webencodings-0.5.1 wrapt-1.14.0 wurlitzer-3.0.2 yapf-0.32.0 zipp-3.7.0

I have also pasted the list of installed packages in case you can figure out if anything is wrong.

Peter

**Comment by pirlite2** [comment]
Just as a further update, I have a colleague using Linux Mint 20.2 and Spyder 5.2.1 (*cf.* my Linux version 20.3 and Spyder 5.2.2)  and for whom Spyder works. The only differences in the dependencies are:

cloudpickle >=0.5.0           :  1.3.0  ,cloudpickle >=0.5.0 : 2.0.0  
IPython >=7.6.0               :  8.0.0  ,IPython >=7.6.0;<8.0.0 : 7.32.0  
jedi >=0.17.2;<0.19.0         :  0.18.0  ,jedi >=0.17.2;<0.19.0 : 0.17.2  
jsonschema >=3.2.0            :  3.2.0  ,jsonschema >=3.2.0 : 4.4.0  
nbconvert >=4.0               :  5.6.1  ,nbconvert >=4.0 : 6.4.4  
numpydoc >=0.6.0              :  0.7.0  ,numpydoc >=0.6.0 : 1.2  
parso >=0.7.0;<0.9.0          :  0.8.2  ,parso >=0.7.0;<0.9.0 : 0.7.1  
pygments >=2.0                :  2.3.1  ,pygments >=2.0 : 2.11.2  
pylint >=2.5.0                :  2.9.6  ,pylint >=2.5.0 : 2.12.2  
pylsp_black >=1.0.0           :  1.0.0  ,pylsp_black >=1.0.0 : 1.1.0  
qtawesome >=1.0.2             :  1.0.3  ,qtawesome >=1.0.2 : 1.1.1  
qtpy >=1.5.0                  :  1.9.0  ,qtpy >=1.5.0 : 2.0.1  
setuptools >=49.6.0           :  58.0.4  ,setuptools >=49.6.0 : 60.10.0  
sphinx >=0.6.6                :  1.8.5  ,sphinx >=0.6.6 : 4.4.0  
textdistance >=4.2.0          :  4.2.1  ,textdistance >=4.2.0 : 4.2.2  
watchdog >=0.10.3             :  2.1.5  ,watchdog >=0.10.3 : 2.1.6  

The first column shows the older, working version whereas the right column shows the newer but non-working version. All other packages -- I have deleted these lines from the above listing for clarity -- have the same version numbers. 

The thing that surprises me is that the version numbers in my more recently downloaded (but non working) Spyder are *lower* than for my colleague's older but working Spyder for: iPython, jedi and parso. It seems odd to 'downgrade' that way?

Not sure if this helps.

Peter

**Comment by ccordoba12** [comment]
> FWIW: 'sudo pip3 install spyder' doesn't produce any change either.

Please note that I didn't ask you to do that, I only asked you what method you used to install Spyder. This is a terrible practice because it mixes the packages provided by your Linux distro with the ones in PyPI, which can break your system very very easily.

To fix that problem, please remove all packages that are installed in this directory

    /usr/local/lib/python3.8

using `sudo pip uninstall` and then remove that directory.

If your system starts to fail randomly after that, I'm afraid you'll have to reinstall your distro.

> No virtual environments. Just the default.

This is (again) not correct because it mixes your distro packages with PyPI ones, but it's not as bad as the previous case. To fix this, please remove all packages installed in this directory

    ~/.local/lib/python3.8

and then the directory itself.

----

After fixing your system with the previous two steps, please create a virtualenv and install Spyder there. Let us know if it works for you.

**Comment by pirlite2** [comment]
>>     FWIW: 'sudo pip3 install spyder' doesn't produce any change either.

>Please note that I didn't ask you to do that, I only asked you what method you used to install Spyder. This is a terrible practice because it mixes the packages provided by your Linux distro with the ones in PyPI, which can break your system very very easily.

OK. Because I had already posted exactly how I was installing, I misunderstood that you were suggesting I use 'sudo'. 

I have followed the suggested clean-up steps, installed Spyder in a virtual environment... and it works! Thanks.

A couple of corollaries though:

1. Having re-read the installation instructions on the website, I can see that I failed to appreciate the significance of creating a virtual environment. Can I suggest the wording is changed from "You’ll first want to create and activate a virtual environment in which to install Spyder, via one of the following methods." to something like "You will **need** to create and activate a virtual environment in which to install Spyder, via one of the following methods." At present, it reads as if it's optional but it looks like it's obligatory.
2. Although I have a lot of experience with C(++) development, I am newish to Python - I need to use s/w that is only available via PyPi. That things can seemingly break on something as small as an increment in patch number is a real culture shock!  (My colleague managed to get Spyder 5.2.1 working without any messing with virtual environments whereas my attempts to follow his steps with v5.2.2 fell down a hole. Hmmmm...)

Anyways, now working. Thanks again.
